broker/core/express-commons/src/middleware/index.js

```diff
@@ -35,6 +36,35 @@ exports.basicAuth = function (username, password) {
   };
 };
 
+exports.tlsAuth = function () {
+  return function (req, res, next) {
+    const clientCertificate = req.headers['ssl-client-cert'];
+    if (!clientCertificate) {
+      logger.error('clientCertificate not found in the request headers: ', req.headers);
+      next(new Unauthorized());
+    } else {
+      const smCertSubjectPattern = config.smConnectionSettings.sm_certificate_subject_pattern;
+      if (smCertSubjectPattern) {
+        let subjectDN = req.headers['ssl-client-subject-dn'];
+        if (!subjectDN || !verifySubjectDN(subjectDN, smCertSubjectPattern)) {
+          logger.error('subject DN doesnt match the sm_certificate_subject_pattern, subjectDN: ', subjectDN, ' sm_certificate_subject_pattern: ', smCertSubjectPattern);
+          return next(new Unauthorized());
+        }
+        next();
+      } else {
+        return next(new Unauthorized());
```

Comment:
Hi @Pooja-08 , a logger before return statement with message stating expected subject pattern not defined would be good to have. Since this is a config parameter, there are high changes teams will forget to add and we might get generic unAuthorized error.